src/reference/v8.json

```diff
@@ -2326,11 +2326,10 @@
       "property-type": "data-driven"
     },
     "text-offset": {
-      "type": "array",
-      "doc": "Offset distance of text from its anchor. Positive values indicate right and down, while negative values indicate left and up. If used with text-variable-anchor, input values will be taken as absolute values. Offsets along the x- and y-axis will be applied automatically based on the anchor position.",
+      "type": "offsetCollection",
+      "doc": "Offset distance of text from its anchor. Positive values indicate right and down, while negative values indicate left and up. \n\n If used with `text-variable-anchor`, a single point or array containing a single point used for `text-offset` will be taken as absolute values. Offsets along the x- and y-axis will be applied automatically based on the anchor position. \n\n If an array of points is provided for `text-offset`, the number of points should match the number of entries provided via `text-variable-anchor`. Each point in the array is used as the offset distance of text from its anchor when the corresponding anchor is used by the renderer. If more entries are provided for `text-variable-anchor` than `text-offset`, the last value in `text-offset` will be used.",
```

Comment:
I didn't understand the part about 
`text-variable-anchor` isn't text-offset enough for placing the labels in different locations?

Reply:
No; they work together. `text-variable-anchor` gives the possible slots to try when placing the anchor. `text-offset` provides the amount to shift the text when one of those slots is considered.

Reply:
Ahh, I think I misinterpreted the design you wrote :-(
I'll re-read it and add my comments there, I have a gut feeling this is better of with a new property - something like text-offset-for-variable-anchors or something, but I need to re-read... I'll keep you posted soon.

Reply:
We did have a discussion on "new property" vs "update existing" for this back in November. Opinions can change, of course..

Reply:
True, I want to re-read those again, sorry for the circular conversation...

Reply:
I thought about it a bit more last night. A new property just for the variable anchor offset could make the implementation and documentation simpler, though we'd then have three properties (text-offset, text-radial-offset, text-variable-offset) that are variations on the same thing.

Reply:
For whats it worth, I would expect something else under text-variable-offset, that is a list of possible offsets the placer should try (analog but *orthogonal* to text-variable-anchor). This could be a useful separate feature in the future and I would recommend to therefore not use that property name.

Reply:
Harel suggested `text-offset-for-variable-anchors` as an off-the-cuff alternative. That's a little verbose for me. Open to suggestions for something in between.

Reply:
It just occurred to me that this feature would also be useful if the text-anchor is set by an expression and the current design does not help there. given that, i would either use text-anchor-offsets but with a fixed order independent of text-variable-anchors so it also works with a text-anchor expression, or separate properties text-offset-left, text-offset-right, text-offset-center etc for each possible anchor setting. perhaps with this in mind its better again to just expand text-offset, as now the documentation (and presumably also the implementation) would actually be independent of text-variable-anchor and just use the actual anchor no matter how this anchor was set.